File: 240326/BOJ-2839.cpp
#include <iostream>

using namespace std;

int main(int argc, char** argv){
    int N;
    cin >> N;

    int answer = -1;
    int standard = N / 5;
    for(int i=standard; i>=0; i--){
        int cnt = i;
        int left = N - 5*i;
        if(left % 3 == 0){
            cnt += left / 3;
            answer = cnt;
            break;
        }
    }

    cout << answer;
}

File: 240326/BOJ-2839.py
N = int(input())

answer = -1
standard = N // 5
for i in range(standard,-1,-1):
    cnt = i
    left = N - 5 * i
    if left % 3 == 0:
        cnt += left // 3
        answer = cnt
        break

print(answer)